staging/src/kubevirt.io/client-go/api/v1/schema.go

```diff
@@ -492,6 +492,17 @@ type Disk struct {
 	// If specified, disk address and its tag will be provided to the guest via config drive metadata
 	// +optional
 	Tag string `json:"tag,omitempty"`
+	// If specified, the virtual disk will be presented with the given block sizes.
+	// +optional
+	BlockIO *BlockIO `json:"blockSize,omitempty"`
```

Comment:
what's the use case for someone needing to explicitly set these physical/logical values rather than using the auto detected values that match the underlying storage?

Reply:
You might want to simulate a 4096n device on top of a 512n or 512e device, or if you have some legacy thing that can't handle 4096 size devices you can force it to 512n or 512e.  
Only relying on the automatic detection also blocks out the option to just stick to 512n as it currently does (assuming it's not put behind a feature flag) and may complicate things if you end up upgrading the backing devices to 4096n later.

Reply:
> it's not put behind a feature flag) and may complicate things if you end up upgrading the backing devices to 4096n later.

i see, so it ensures that the logical/physical block sizes remain consistent regardless of what happens to the storage that the vm disk resides on.

Reply:
Pretty much. The exact behaviour/effect is related to the other comment you made.

Under the assumption that auto-detection would be enabled either by default or a feature flag, setting these would ensure the VM doesn't suddenly change. 
As mentioned in your other comment, the idea to be able to opt-in per VM sounds a lot better to me and doesn't require you to update all your VM configs if you wanted to enable the feature flag for it without potentially impacting all VMs.